app/assets/javascripts/components/NotesView.tsx

```diff
@@ -0,0 +1,256 @@
+import { NotesList } from './NotesList';
+import { NotesListOptionsMenu } from './NotesListOptionsMenu';
+import { PanelResizer } from './PanelResizer';
+import { SearchOptions } from './SearchOptions';
+import { toDirective } from './utils';
+
+type Props = {
+  application: WebApplication;
+  appState: AppState;
+};
+
+const NotesView: FunctionComponent<Props> = observer(
+  ({ application, appState }) => {
+    const notesViewPanelRef = useRef<HTMLDivElement>(null);
+
+    const {
+      completedFullSync,
+      createNewNote,
+      displayOptions,
+      noteFilterText,
+      optionsSubtitle,
+      panelTitle,
+      renderedNotes,
+      selectedNotes,
+      setNoteFilterText,
+      showDisplayOptionsMenu,
+      toggleDisplayOptionsMenu,
+      searchBarElement,
+      selectNextNote,
+      selectPreviousNote,
+      onFilterEnter,
+      handleFilterTextChanged,
+      onSearchInputBlur,
+      clearFilterText,
+      paginate,
+    } = appState.notesView;
+
+    useEffect(() => {
+      handleFilterTextChanged();
+    }, [noteFilterText, handleFilterTextChanged]);
+
+    useEffect(() => {
+      /**
+       * In the browser we're not allowed to override cmd/ctrl + n, so we have to
+       * use Control modifier as well. These rules don't apply to desktop, but
+       * probably better to be consistent.
+       */
+      const newNoteKeyObserver = application.io.addKeyObserver({
+        key: 'n',
+        modifiers: [KeyboardModifier.Meta, KeyboardModifier.Ctrl],
+        onKeyDown: (event) => {
+          event.preventDefault();
+          createNewNote();
+        },
+      });
+
+      const nextNoteKeyObserver = application.io.addKeyObserver({
+        key: KeyboardKey.Down,
+        elements: [
+          document.body,
+          ...(searchBarElement ? [searchBarElement] : []),
+        ],
+        onKeyDown: () => {
+          if (searchBarElement === document.activeElement) {
+            searchBarElement?.blur();
+          }
+          selectNextNote();
+        },
+      });
+
+      const previousNoteKeyObserver = application.io.addKeyObserver({
+        key: KeyboardKey.Up,
+        element: document.body,
+        onKeyDown: () => {
+          selectPreviousNote();
+        },
+      });
+
+      const searchKeyObserver = application.io.addKeyObserver({
+        key: 'f',
+        modifiers: [KeyboardModifier.Meta, KeyboardModifier.Shift],
+        onKeyDown: () => {
+          if (searchBarElement) {
+            searchBarElement.focus();
+          }
+        },
+      });
+
+      return () => {
+        newNoteKeyObserver();
+        nextNoteKeyObserver();
+        previousNoteKeyObserver();
+        searchKeyObserver();
+      };
+    }, [
+      application.io,
+      createNewNote,
+      searchBarElement,
+      selectNextNote,
+      selectPreviousNote,
+    ]);
+
+    const onNoteFilterTextChange = (e: Event) => {
+      setNoteFilterText((e.target as HTMLInputElement).value);
+    };
```

Comment:
For what it's worth,
there are types for these, they are tricky to find, but they make your code fully type checked,

https://github.com/standardnotes/web/blob/develop/app/assets/javascripts/components/TagsListItem.tsx#L45